experimental/packages/otlp-exporter-base/src/transport/http-exporter-transport.ts

```diff
@@ -56,23 +56,20 @@ class HttpExporterTransport implements IExporterTransport {
     // intentionally left empty, nothing to do.
   }
 
-  private _loadUtils(): Utils {
+  private async _loadUtils(): Promise<Utils> {
     let utils = this._utils;
 
     if (utils === null) {
-      // Lazy require to ensure that http/https is not required before instrumentations can wrap it.
-      const {
-        sendWithHttp,
-        createHttpAgent,
-        // eslint-disable-next-line @typescript-eslint/no-require-imports
-      } = require('./http-transport-utils');
+      // Lazy import to ensure that http/https is not required before instrumentations can wrap it.
+      const imported = await import('./http-transport-utils.js');
 
       utils = this._utils = {
-        agent: createHttpAgent(
-          this._parameters.url,
-          this._parameters.agentOptions
+        agent: await this._parameters.agent(
+          new URL(this._parameters.url).protocol
```

Comment:
Should we add some exception handling here as `new URL()` will throw an exception if input is invalid.